app/version/version.go

```diff
@@ -57,25 +59,122 @@ func GitCommit() (hash string, timestamp string) {
 func LogInfo(ctx context.Context, msg string) {
 	gitHash, gitTimestamp := GitCommit()
 	log.Info(ctx, msg,
-		z.Str("version", Version),
+		z.Str("version", Version.String()),
 		z.Str("git_commit_hash", gitHash),
 		z.Str("git_commit_time", gitTimestamp),
 	)
 }
 
-// Minor returns the minor version of the provided version string.
-func Minor(version string) (string, error) {
-	split := strings.Split(version, ".")
-	if len(split) < 2 {
-		return "", errors.New("invalid version string")
+type semVerType int
+
+const (
+	typeMinor semVerType = iota
+	typePatch
+	typePreRelease
+)
+
+// SemVer represents a semantic version. A valid SemVer contains a major and minor version
+// and optionally either a typePatch version or a pre-release label, i.e., v1.2 or v1.2.3 or v1.2-rc.
+type SemVer struct {
```

Comment:
because we do not follow SemVer exactly, the `PreRelease` tag vs minor vs patch is our custom logic. And it is super simple, so not wroth a dependency